### PR TITLE
OpInfo: log_softmax

### PR DESCRIPTION
**Pull request by kshitij12345**
Reference: #54261

### CODE REVIEW

File: torch/testing/_internal/common_methods_invocations.py [hunk 1]
```diff
@@ -6836,6 +6852,22 @@ def gradcheck_wrapper_triangular_input(op, input, *args, upper=False, **kwargs):
...
+        supports_out=False,
+        dtypes=floating_types_and(torch.bfloat16),
+        dtypesIfCUDA=floating_types_and(torch.float16, torch.bfloat16),
+        sample_inputs_func=sample_inputs_log_softmax,
+        assert_autodiffed=True),
+    OpInfo(
+        'log_softmax',
+        variant_test_name='dtype',
+        supports_out=False,
+        dtypes=all_types_and_complex_and(torch.bool, torch.float16, torch.bfloat16),
```
Comment (kshitij12345) [inline torch/testing/_internal/common_methods_invocations.py:6868]: Not sure, as to how complex is also supported with `dtype=torch.float64`.

Comment (kshitij12345) [reply torch/testing/_internal/common_methods_invocations.py:6868]: Looks like this part of code facilitates that,
https://github.com/pytorch/pytorch/blob/c3bf42e0d8b0a820a0fc762e520de4a6377562a3/aten/src/ATen/native/SoftMax.cpp#L273-L274

```python
>>> t = torch.randn(3,3, dtype=torch.complex128)
>>> torch.log_softmax(t, 0, dtype=torch.float64)
<stdin>:1: UserWarning: Casting complex values to real discards the imaginary part (Triggered internally at  ../aten/src/ATen/native/Copy.cpp:240.)
```

It does raise a warning. But I think it should just error out.

Comment (mruberry) [reply torch/testing/_internal/common_methods_invocations.py:6868]: This is really interesting -- thanks for taking a closer look. I don't think we should take any action for this behavior in this PR, however.

fyi @anjali411

**Review by mruberry — APPROVED** [review]
Wahoo!
